tests/Codec/ObjectCodecTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace MongoDB\Tests\Codec;
+
+use MongoDB\Codec\Codec;
+use MongoDB\Codec\CodecLibrary;
+use MongoDB\Codec\DecodeIfSupported;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Codec\ObjectCodec;
+use MongoDB\Exception\UnsupportedValueException;
+use MongoDB\Tests\TestCase;
+use stdClass;
+
+class ObjectCodecTest extends TestCase
+{
+    public function testDecodeObject(): void
```

Comment:
Is it feasible to use data providers here? The common code looks to be:

```
$this->assertEquals($expected, $this->getCodec()->decode($value));
```

Reply:
Updated.